### PR TITLE
Doha - API Gateway in Terraform

### PR DESCRIPTION
**Pull request by DewDropsDoha**
Fixes #174 

- [ ] Does not depend on other PRs
- [x] Depends on other PR being merged first: #200

## What kind of change does this PR introduce (check at least one)

- [ ] Bugfix
- [ ] Improvement
- [x] New feature
- [ ] Refactor
- [ ] Infrastructure
- [ ] Other (please specify)

## What was done
API Gateway (HttpApi) implemented and generated base_url working fine.
Refs: https://learn.hashicorp.com/tutorials/terraform/lambda-api-gateway?_ga=2.267865927.81444029.1661240980-1436054164.1661240980#create-an-http-api-with-api-gateway

## How to test

- [ ] Needs regression testing (add QA needed label)

## Deploy risk & follow up

Please assess risk & follow up on deployment of this PR. Elaborate your choice if selecting anything other than 'No risk'.

- [x] No risk
- [ ] Low
- [ ] High

QA - if regression testing was requested:

- [ ] QA has verified

### CODE REVIEW

File: doha-aws-node-project/terraform/terraform.tfstate [hunk 1]
```diff
@@ -1,9 +1,13 @@
 {
```
[the remark this answers is not in the document]
Comment (DewDropsDoha) [reply doha-aws-node-project/terraform/terraform.tfstate:1]: tfstate files added in gitignore. Commit 8c3b2c8

File: doha-aws-node-project/terraform/terraform.tfstate.backup [hunk 2]
```diff
@@ -1,9 +1,13 @@
 {
   "version": 4,
   "terraform_version": "1.2.7",
-  "serial": 53,
+  "serial": 62,
```
[the remark this answers is not in the document]
Comment (DewDropsDoha) [reply doha-aws-node-project/terraform/terraform.tfstate.backup:4]: tfstate files added in gitignore. Commit 8c3b2c8

File: doha-aws-node-project/terraform/terraform.tfstate (hunk 1 above)
[the remark this answers is not in the document]
Comment (mhazad) [reply doha-aws-node-project/terraform/terraform.tfstate]: And then you need to delete the file from git.

File: doha-aws-node-project/terraform/terraform.tfstate.backup (hunk 2 above)
[the remark this answers is not in the document]
Comment (mhazad) [reply doha-aws-node-project/terraform/terraform.tfstate.backup]: And then you need to delete the file from git.

**Review by mhazad — CHANGES_REQUESTED** [review]
Pls, remove unnecessary files.

File: doha-aws-node-project/terraform/terraform.tfstate (hunk 1 above)
[the remark this answers is not in the document]
Comment (DewDropsDoha) [reply doha-aws-node-project/terraform/terraform.tfstate]: tfstate files removed. Commit cdb03c0

File: doha-aws-node-project/terraform/terraform.tfstate.backup (hunk 2 above)
[the remark this answers is not in the document]
Comment (DewDropsDoha) [reply doha-aws-node-project/terraform/terraform.tfstate.backup]: terraform.tfstate file removed. Commit cdb03c0

**Pull request merged**